code/datums/quirks/neutral_quirks/transhumanist.dm

```diff
@@ -100,27 +101,54 @@
-	var/limb_type = GLOB.limb_choice_transhuman[client_source?.prefs?.read_preference(/datum/preference/choiced/prosthetic)]
-	if(isnull(limb_type))  //Client gone or they chose a random prosthetic
-		limb_type = GLOB.limb_choice_transhuman[pick(GLOB.limb_choice_transhuman)]
+	var/part_type = GLOB.part_choice_transhuman[client_source?.prefs?.read_preference(/datum/preference/choiced/trans_prosthetic)]
+	if(isnull(part_type))  //Client gone or they chose a random part
+		part_type = GLOB.part_choice_transhuman[pick(GLOB.part_choice_transhuman)]
 
 	var/mob/living/carbon/human/human_holder = quirk_holder
-	var/obj/item/bodypart/new_part = new limb_type()
-
-	slot_string = "[new_part.plaintext_zone]"
-	old_limb = human_holder.return_and_replace_bodypart(new_part, special = TRUE)
+	var/obj/item/new_part = new part_type()
+	if(isbodypart(new_part))
+		var/obj/item/bodypart/new_bodypart = new_part
+		slot_string = "[new_bodypart.plaintext_zone]"
+		old_part = human_holder.return_and_replace_bodypart(new_bodypart, special = TRUE)
+	else if(isorgan(new_part))
+		var/obj/item/organ/new_organ = new_part
+		old_part = human_holder.get_organ_slot(new_organ.slot)
+		if(new_organ.Insert(human_holder, special = TRUE))
+			old_part.moveToNullspace()
+			STOP_PROCESSING(SSobj, old_part)
+			slot_string = "[new_organ.name]"
 
 /datum/quirk/transhumanist/post_add()
 	if(slot_string)
-		to_chat(quirk_holder, span_boldannounce("Your [slot_string] has been replaced with a robot arm. You need to use a welding tool and cables to repair it, instead of sutures and regenerative meshes."))
+		if(isbodypart(old_part))
+			to_chat(quirk_holder, span_boldannounce("Your [slot_string] has been replaced with a robotic limb. You need to use a welding tool and cables to repair it, instead of sutures and regenerative meshes."))
+		else if (old_part.name == "eyes")
+			to_chat(quirk_holder, span_boldannounce("You replaced your eyes with flashlights, not cameras. You can't see a thing!"))
+		else if (isorgan(old_part))
+			to_chat(quirk_holder, span_boldannounce("Your [slot_string] brings you one step closer to silicon perfection, but you feel you're not quite there yet."))
 
 /datum/quirk/transhumanist/remove()
-	if(old_limb)
+	if(old_part)
 		var/mob/living/carbon/human/human_holder = quirk_holder
-		human_holder.del_and_replace_bodypart(old_limb, special = TRUE)
-		old_limb = null
+		if(isbodypart(old_part))
+			var/obj/item/bodypart/old_bodypart = old_part
+			human_holder.del_and_replace_bodypart(old_bodypart, special = TRUE)
+			old_bodypart = null
+		else if(isorgan(old_part))
+			var/obj/item/organ/old_organ = old_part
+			old_part = human_holder.get_organ_slot(ORGAN_SLOT_TONGUE)
+			old_organ.Insert(quirk_holder, special = TRUE)
+			old_part.moveToNullspace()
+			STOP_PROCESSING(SSobj, old_part)
+			old_organ = null
+			old_part = null
+
 	quirk_holder.clear_mood_event(MOOD_CATEGORY_TRANSHUMANIST_BODYPART)
 	quirk_holder.clear_mood_event(MOOD_CATEGORY_TRANSHUMANIST_PEOPLE)
```

Comment:
invert this into an early return, so something like:

```dm
/datum/quirk/transhumanist/remove()
 if(isnull(old_limb) // isnull() instead of ! because it's more explicit (and also slightly faster)
  // clear the mood events
  return

 // now add all of the new code with bodyparts/organs here that was previously in the if() conditional
```

this is just to cut down on how ugly the indentation gets here.

Reply:
The early returns should be in now, the build got mad at me when I removed the stringification so I guess that stays, thank you for the suggestions